course/utils.py

```diff
@@ -659,10 +660,25 @@ def has_permission(self, perm, argument=None):
         else:
             return (perm, argument) in self.permissions()
 
+    def _set_course_lang(self, action):
+        # type: (Text) -> None
+        from django.conf import settings
+        if not getattr(settings, "RELATE_ENABLE_COURSE_SPECIFIC_LANG", False):
+            return
+        if self.course.force_lang is not None:
+            if action == "activate":
+                self.old_language = translation.get_language()
```

Comment:
This should assert that `old_language` had not previously been set, to avoid incorrect behavior if this context manager is used in a nested manner.

Reply:
Do you mean
```python
if action == "activate" and self.old_language is not None?
```

Reply:
No. I meant that

* if we don't expect nested use: this should crash if `old_language` is already set (requires that `old_language` be initialized in the constructor),
* if we do expect nested use: use a list/stack of `old_languages` instead.

IOW, have a definite answer for what should happen if use of this context mgr is nested.

Reply:
Then it can be view as a good side effect to check whether the `course_view` is nestedly (duplicated) used, for developers, am I right? Then why not add another property such as `_in_context_manager` to indicate that?

Reply:
That's fine, too.